06_puppy/alextmz/mapstore.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"errors"
+)
+
+type MapStore map[uint16]*Puppy
+
+func NewmapStore() *MapStore {
+	a := MapStore{}
+	return &a
+}
+
+func (m *MapStore) CreatePuppy(p *Puppy) error {
+	if _, ok := (*m)[p.ID]; !ok {
+		(*m)[p.ID] = p
+		return nil
+	}
+	return errors.New("puppy ID being created already exists")
+}
+
+func (m *MapStore) ReadPuppy(id uint16) (*Puppy, error) {
+	if v, ok := (*m)[id]; ok {
+		return v, nil
+	}
+	return nil, errors.New("puppy ID being read does not exist")
+}
+
+func (m *MapStore) UpdatePuppy(id uint16, p *Puppy) error {
+	if _, ok := (*m)[id]; ok {
+		(*m)[id] = p
+		return nil
+	}
+	return errors.New("puppy ID being updated does not exist")
```

Comment:
same as above more info in error message.

Reply:
suggestion :you might have to handle the case when id !=p.Id as an exception or call out the assumption that Update overrides a puppy with given id with that provided by user.